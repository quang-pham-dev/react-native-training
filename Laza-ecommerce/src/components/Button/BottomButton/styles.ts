import { StyleSheet } from 'react-native';
// Styles
import { Metrics, Colors, Fonts } from '@styles/themes/index';

const styles = StyleSheet.create({
  button: {
    paddingVertical: Metrics.padding.md,
    paddingHorizontal: Metrics.padding.xl,
    alignItems: 'center',
  },
  title: {
    color: Colors.white,
    textTransform: 'none',
    fontFamily: Fonts.fontFamily.Inter_600SemiBold,
    fontSize: Fonts.size.sm,
    lineHeight: Fonts.lineHeight.lg,
  },
});

export default styles;
