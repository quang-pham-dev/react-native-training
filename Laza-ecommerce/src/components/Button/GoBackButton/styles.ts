// Styles
import { Metrics } from '@styles/themes/index';
import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  button: {
    marginTop: Metrics.margin.customXxl,
    marginLeft: Metrics.margin.lg,
    alignItems: 'center',
  },
  icon: {
    width: Metrics.icons.semiLarge,
    height: Metrics.icons.semiLarge,
  },
});

export default styles;
