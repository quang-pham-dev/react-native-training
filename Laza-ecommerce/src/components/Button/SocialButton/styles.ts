import { StyleSheet } from 'react-native';
// Styles
import { Metrics, Colors, Fonts } from '@styles/themes/index';

const styles = StyleSheet.create({
  button: {
    paddingVertical: Metrics.padding.md,
    paddingHorizontal: Metrics.padding.xl,
    alignItems: 'center',
    borderColor: Colors.transparent,
    borderWidth: 1,
    backgroundColor: Colors.white,
  },
  title: {
    color: Colors.white,
    textTransform: 'none',
    fontFamily: Fonts.fontFamily.Inter_600SemiBold,
    fontSize: Fonts.size.sm,
    lineHeight: Metrics.height.custom,
  },
  icon: {
    width: Metrics.icons.custom,
    height: Metrics.icons.custom,
  },
});

export default styles;
