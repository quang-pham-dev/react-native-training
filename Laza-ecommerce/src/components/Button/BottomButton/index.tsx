import React, { memo } from 'react';
import { Text, TouchableOpacity } from 'react-native';
// Types
import { BottomButtonProps } from 'types/Button';
// Styles
import styles from './styles';

const BottomButton = ({
  title,
  disabled,
  buttonStyles = [],
  children = [],
  onPress = () => {},
}: BottomButtonProps) => {
  return (
    <TouchableOpacity
      style={[styles.button, buttonStyles]}
      onPress={onPress}
      disabled={disabled}
      activeOpacity={0.8}
    >
      <Text style={[styles.title]}>{title}</Text>
      {children}
    </TouchableOpacity>
  );
};

export default memo(BottomButton);
