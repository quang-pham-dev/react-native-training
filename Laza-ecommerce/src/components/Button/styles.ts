import { StyleSheet } from 'react-native';
import { Colors } from 'styles/themes';

const styles = StyleSheet.create({
  button: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  icon: {},
  text: {
    color: Colors.white,
  },
});
export default styles;
