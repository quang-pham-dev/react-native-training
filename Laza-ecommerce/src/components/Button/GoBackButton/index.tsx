import React, { memo } from 'react';
import { Image, TouchableOpacity } from 'react-native';
// Types
import { GoBackButtonProps } from 'types/Button';
// Styles
import styles from './styles';

const GoBackButton = ({
  icon,
  disabled,
  buttonStyles = [],
  onPress = () => {},
}: GoBackButtonProps) => {
  return (
    <TouchableOpacity
      style={[styles.button, buttonStyles]}
      onPress={onPress}
      disabled={disabled}
      activeOpacity={0.8}
    >
      {icon && <Image style={styles.icon} source={icon} />}
    </TouchableOpacity>
  );
};

export default memo(GoBackButton);
