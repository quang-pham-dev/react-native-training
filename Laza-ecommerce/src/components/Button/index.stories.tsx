import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react-native';
import React from 'react';
import CenterView from '../../../storybook/stories/CenterView';
import Colors from 'themes/Colors';
import IMAGES from 'themes/Images';
import Button from './index';

storiesOf('Button', module)
  .addDecorator(getStory => <CenterView>{getStory()}</CenterView>)
  .add('Circle type ', () => (
    <Button onPress={action('clicked')} type='Circle' icon={IMAGES.iconMenu} />
  ))
  .add('Bottom type ', () => <Button text='Login' onPress={action('clicked')} type='Bottom' />)
  .add('Social type ', () => (
    <Button
      text='Google'
      onPress={action('clicked')}
      icon={IMAGES.iconGoogle}
      type='Social'
      backgroundColor={Colors.google}
    />
  ));
