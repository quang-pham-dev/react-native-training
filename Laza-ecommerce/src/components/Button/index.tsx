import React from 'react';
import { Image, Text, TouchableOpacity } from 'react-native';
// Types
import { ButtonProps } from 'types/Button';
// Styles
import styles from './styles';

const Button = ({
  onPress = () => {},
  text,
  textStyles,
  buttonStyles,
  icon,
  iconStyles,
  disabled,
  children,
  ...props
}: ButtonProps) => {
  return (
    <TouchableOpacity
      style={[styles.button, buttonStyles]}
      {...props}
      onPress={onPress}
      disabled={disabled}
      activeOpacity={0.8}
    >
      {icon && <Image style={[styles.icon, iconStyles]} source={icon} />}
      {text && <Text style={[styles.text , textStyles]}>{text}</Text>}
      {children}
    </TouchableOpacity>
  );
};

export default Button;
