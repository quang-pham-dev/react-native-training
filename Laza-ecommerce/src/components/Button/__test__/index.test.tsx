import React from 'react';
import renderer from 'react-test-renderer';
import { IMAGES } from 'styles/themes';
import Button from 'components/Button';

describe('Button Component', () => {
  test('should render correctly', () => {
    const tree = renderer.create(<Button text='Facebook' onPress={jest.fn()} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render icon button correctly', () => {
    const tree = renderer
      .create(<Button text='Google' icon={IMAGES.iconGoogle} onPress={jest.fn()} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
