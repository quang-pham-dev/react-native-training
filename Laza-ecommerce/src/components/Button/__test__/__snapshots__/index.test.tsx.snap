// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
        },
        undefined,
      ]
    }
  >
    Facebook
  </Text>
</View>
`;

exports[`Button Component should render icon button correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
        },
        undefined,
      ]
    }
  >
    Google
  </Text>
</View>
`;
